packages/test/test-end-to-end-tests/src/test/gc/gcSweepAttachmentBlobs.spec.ts

```diff
@@ -0,0 +1,1063 @@
+				// this container does not have access to the blob's handle.
+				await validateBlobRetrievalFails(
+					container2,
+					blobHandle1.absolutePath,
+					"Container2: Blob1",
+				);
+				await validateBlobRetrievalFails(
+					container2,
+					blobHandle2.absolutePath,
+					"Container2: Blob2",
+				);
+
+				// Retrieving the blobs in the summarizer container should fail as well.
+				await validateBlobRetrievalFails(
+					summarizerContainer,
+					blobHandle1.absolutePath,
+					"Summarizer: Blob1",
+				);
+				await validateBlobRetrievalFails(
+					summarizerContainer,
+					blobHandle2.absolutePath,
+					"Summarizer: Blob2",
+				);
+			},
+		);
+
+		itExpects(
+			"deletes blobs uploaded and de-duped in disconnected container",
+			[
+				{
+					eventName: "fluid:telemetry:BlobManager:GC_Deleted_Blob_Requested",
+					clientType: "interactive",
+				},
+				{
+					eventName: "fluid:telemetry:BlobManager:GC_Deleted_Blob_Requested",
+					clientType: "interactive",
+				},
+				{
+					eventName: "fluid:telemetry:BlobManager:GC_Deleted_Blob_Requested",
+					clientType: "interactive",
+				},
+				{
+					eventName: "fluid:telemetry:BlobManager:GC_Deleted_Blob_Requested",
+					clientType: "noninteractive/summarizer",
+				},
+			],
+			async () => {
+				const { mainContainer, mainDataStore } = await createContainerAndDataStore();
+
+				const { summarizer, summarizerContainer } =
+					await createSummarizerWithInitialSummary(mainContainer);
+
+				// Disconnect the main container, upload couple of blobs with same content and mark them referenced. When
+				// these blobs are uploaded to the server, they will be de-duped and redirect to the same storageId.
+				mainContainer.disconnect();
+				const blobContents = "Blob contents";
+				const blobHandle1 = await mainDataStore._context.uploadBlob(
+					stringToBuffer(blobContents, "utf-8"),
+				);
+				const blobHandle2 = await mainDataStore._context.uploadBlob(
+					stringToBuffer(blobContents, "utf-8"),
+				);
+
+				// Connect the container after the blob is uploaded. Send an op to transition the container to write mode.
+				mainContainer.connect();
+				mainDataStore._root.set("transition to write", "true");
+				await ensureContainerConnectedWriteMode(mainContainer);
+
+				// Add the blob handles to reference them.
+				mainDataStore._root.set("blob1", blobHandle1);
+				mainDataStore._root.set("blob2", blobHandle2);
+
+				// Upload the same blob. This will get de-duped and we will get back another handle.
+				const blobHandle3 = await mainDataStore._context.uploadBlob(
+					stringToBuffer(blobContents, "utf-8"),
+				);
+				assert.notStrictEqual(
+					blobHandle1.absolutePath,
+					blobHandle3.absolutePath,
+					"blob handles should be different",
+				);
+
+				// Add the new handle and then remove all the handles to unreference the blob.
+				mainDataStore._root.set("blob3", blobHandle2);
```

Comment:
Should this be blobHandle3?

Reply:
Yes - https://github.com/microsoft/FluidFramework/pull/14249